backend/src/modules/transactions/transactions.controller.ts

```diff
@@ -65,10 +76,23 @@ export class TransactionsController {
   public async getTransactionByQuery(
     @Body() query: QueryDTO,
     @Res() res: Response,
+    @Req() req: Request,
   ) {
     this.logger.log('POST to /api/transactions/query');
-    const transactions = await this.queryService.query(query);
-    return res.status(HttpStatus.OK).send(transactions);
+    if ('Token' in req.cookies) {
+      const userEmail = JSON.parse(
+        Buffer.from(req.cookies.Token, 'base64').toString(),
+      ).email;
+      const user = await this.userService.findByEmail(userEmail);
+      query.selectors.push({
```

Comment:
Hm, man atrodo kad cia turetu buti private API, ir query servisiukas turet koki metoda, kad nereiketu typint in visu situ hardcoded stringu. Nes dabar jei pasikeis vidinis queryService api ir prades luzinet random vietose kodas (kaip cia). @Andrius-B